src/main/java/ch/uzh/ifi/hase/soprafs21/security/SecurityConfig.java

```diff
@@ -0,0 +1,48 @@
+package ch.uzh.ifi.hase.soprafs21.security;
+
+import ch.uzh.ifi.hase.soprafs21.service.JwtTokenUtil;
+import ch.uzh.ifi.hase.soprafs21.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.http.HttpServletResponse;
+
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+    @Autowired
+    private UserService userService;
+
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.cors()
+                .and().csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and().exceptionHandling()
+                .authenticationEntryPoint((req, rsp, e) -> rsp.sendError(HttpServletResponse.SC_UNAUTHORIZED))
+                .and()
+                .addFilterBefore(new JwtTokenAuthenticationFilter(jwtTokenUtil, userService), UsernamePasswordAuthenticationFilter.class)
+                .authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/users/*","/message","/parks/*","/paths/*").permitAll()
```

Comment:
Do we have to list all methods separately? Perhaps just using something more generic like '/**' ?

Reply:
Wouldn't that also match the login route(s)?

Additionally, do we not need a /v1/ prefix with these paths here if we specify them like this?

Reply:
@KseniaBel I added all the valid paths for application. /** will also work, must not be a problem. 
@bschne I tested that this endpoint 'http://localhost:8080/v1/users/1/conversations' from postman for token authentication and it is working fine without /v1.  And /login route will be ignored for filtering, using below code
@Override
    public void configure(WebSecurity web) throws Exception {
        web.ignoring()
                .antMatchers("/v1/users/login");
    }